### PR TITLE
Upgrade docusaurus to version 3

### PR DESCRIPTION
**Pull request by MilanKovacic**
**Close https://github.com/single-spa/single-spa.js.org/pull/624 first!**

### CODE REVIEW

**Review by internettrans — APPROVED** [review]
I didn't review each file since a lot of it was prettier changes. Did you follow the steps in the [Docusaurus v3 migration guide](https://docusaurus.io/docs/migration/v3)? Are there any differences in how the website functions

File: .github/workflows/build_deploy.yml [hunk 1]
```diff
@@ -16,8 +16,8 @@ jobs:
       - uses: actions/checkout@v3
       - uses: actions/setup-node@v3
         with:
-          node-version: 16
-          cache: 'npm'
+          node-version: 20
```
Comment (MilanKovacic) [inline .github/workflows/build_deploy.yml:19]: Docusaurus 3 does not support node 16.

File: docusaurus.config.js [hunk 2]
```diff
@@ -143,7 +143,6 @@ module.exports = {
         { to: "https://baseplate.cloud", label: "Deploy", position: "right" },
       ],
     },
-    prismTheme: require("prism-react-renderer/themes/vsDark"),
```
Comment (MilanKovacic) [inline docusaurus.config.js:146]: Not sure where this was coming from, but the default looks fine.

**Comment by MilanKovacic** [comment]
> I didn't review each file since a lot of it was prettier changes. Did you follow the steps in the [Docusaurus v3 migration guide](https://docusaurus.io/docs/migration/v3)? Are there any differences in how the website functions

History should be clean now.
Yes, I've followed all of the steps.

1. Node upgraded to 20
2. All packages upgraded to latest versions
3. Fixed breaking change related to versioned docs (prefix is not needed anymore)

Website should function and look the same.

File: versioned_sidebars/version-4.x-sidebars.json [hunk 3]
```diff
@@ -1,12 +1,12 @@
 {
-  "version-4.x/docs": [
```
Comment (internettrans) [inline versioned_sidebars/version-4.x-sidebars.json:2]: Is this something that changed with docusaurus@3?

Comment (MilanKovacic) [reply versioned_sidebars/version-4.x-sidebars.json:2]: Yes, ths is not needed or supported anymore: https://github.com/facebook/docusaurus/pull/9310.

**Review by internettrans — APPROVED** [review]

**Pull request merged**